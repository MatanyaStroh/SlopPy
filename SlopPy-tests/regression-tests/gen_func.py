# have generators NOT yield 'NA' values, so as not to corrupt results

import StringIO

lst = ['1', '2', 'bad', '3', 'bad2', 'bad3', 'bad4', '4']

def foo(l):
  for x in l:
    y = 20 * int(x)
    yield y

s = StringIO.StringIO()
for e in foo(lst):
  print >> s, e

assert s.getvalue() == '20\n40\n60\n80\n'

