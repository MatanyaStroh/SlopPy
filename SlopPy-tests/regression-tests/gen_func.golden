{NA <type 'exceptions.ValueError'> "invalid literal for int() with base 10: 'bad'"
 Traceback (most recent call last):
  File "regression-tests/gen_func.py", line 9, in foo
    y = 20 * int(x)

Locals: {'y': 40, 'x': 'bad', 'l': ['1', '2', 'bad', '3', 'bad2', 'bad3', 'bad4', '4']}
}
{binary_op(*,NA) | regression-tests/gen_func.py foo:9}
{gen_iternext() yielded NA | regression-tests/gen_func.py <module>:13}
{NA <type 'exceptions.ValueError'> "invalid literal for int() with base 10: 'bad2'"
 Traceback (most recent call last):
  File "regression-tests/gen_func.py", line 9, in foo
    y = 20 * int(x)

Locals: {'y': 60, 'x': 'bad2', 'l': ['1', '2', 'bad', '3', 'bad2', 'bad3', 'bad4', '4']}
}
{binary_op(*,NA) | regression-tests/gen_func.py foo:9}
{gen_iternext() yielded NA | regression-tests/gen_func.py <module>:13}
{PyObject_SetItem(value=NA) | regression-tests/gen_func.py foo:9}
{NA <type 'exceptions.ValueError'> "invalid literal for int() with base 10: 'bad3'"
 Traceback (most recent call last):
  File "regression-tests/gen_func.py", line 9, in foo
    y = 20 * int(x)

Locals: {'y': 60, 'x': 'bad3', 'l': ['1', '2', 'bad', '3', 'bad2', 'bad3', 'bad4', '4']}
}
{PyObject_SetItem(value=NA) | regression-tests/gen_func.py foo:9}
{binary_op(*,NA) | regression-tests/gen_func.py foo:9}
{gen_iternext() yielded NA | regression-tests/gen_func.py <module>:13}
{PyObject_SetItem(value=NA) | regression-tests/gen_func.py foo:9}
{NA <type 'exceptions.ValueError'> "invalid literal for int() with base 10: 'bad4'"
 Traceback (most recent call last):
  File "regression-tests/gen_func.py", line 9, in foo
    y = 20 * int(x)

Locals: {'y': 60, 'x': 'bad4', 'l': ['1', '2', 'bad', '3', 'bad2', 'bad3', 'bad4', '4']}
}
{PyObject_SetItem(value=NA) | regression-tests/gen_func.py foo:9}
{binary_op(*,NA) | regression-tests/gen_func.py foo:9}
{gen_iternext() yielded NA | regression-tests/gen_func.py <module>:13}
